src/io/flutter/inspector/InspectorService.java

```diff
@@ -255,6 +257,80 @@ private void onVmServiceReceived(String streamId, Event event) {
     return invokeServiceMethodDaemonNoGroup(methodName, params);
   }
 
+  /*
+   * We make a best guess for the pub directory based on the the root directory
+   * of the first non artifical widget in the tree.
+  */
+  public CompletableFuture<String> inferPubRootDirectoryIfNeeded(){
```

Comment:
Do we need to return String here?

Reply:
I would continue to return a CompleteableFuture<String> 
could be useful if you want to display what the inferred root project directory was or as an enhancement pair it with what the project directory defined by the user's actual project was. 
A v2 refinement would be to use this directory to rewrite the user's actual open package roots to match the same path scheme. E.g. what happened was the dirs used by the kernel transformer are probably in a directory with a different file prefix.

Reply:
ack.